docs/auth-apis.md

```diff
@@ -0,0 +1,115 @@
+# Auth API Design 
+
+## Cases Auth APIs must satisfy:
+
+- Allow the user to sign up
+- Allow the user to log in 
+- Provide session token for user/refresh token/invalidate token 
+
+## Required APIs/details
+
+Auth to be displayed on the front end: 
+- Sign up
+- Log in 
+- Log out 
+
+Fields required to be returned by the API:
+- `firstName`
+- `lastName`
+- `email`
+- `password`
+- `token`
+- `userId`
+
+### Sign up/Log in/Log out 
+
+As mentioned above, Auth APIs must return these fields: 
+
+- `firstName`
+- `lastName`
+- `email`
+- `password`
+- `token`
+- `userId`
+
+## Sign up
+
+- The HTTP verb used for this should be `POST`. 
+
+- The REST endpoint for these will be `POST /api/auth/signup`
+
+## Log in
+
+- The HTTP verb used for this should be `POST`. 
+
+- The REST endpoint for these will be `POST /api/auth/login`
+
+## Refresh
```

Comment:
I think it would be easier to just refresh the session on every successful authenticated API call by the user, instead of a dedicated refresh endpoint, but would be glad to hear your thoughts on this

Reply:
I like that idea, I know we talked about it before. I'll look at how to implement that!